src/plans.js

```diff
@@ -358,6 +358,17 @@ function fetchNextJob (schema) {
       WHERE state < '${states.active}'
         AND name LIKE $1
         AND startAfter < now()
+        AND NOT (
```

Comment:
This has the side effect of doubling the index reads.  On a busy instance, this will impact performance. I think it may be better to opt into this additional load like we did with the `includeMetadata` option.

Reply:
Ah good call. I was assuming that the WHERE clause would "short circuit" in non-singleton queue scenarios, but yeah I don't think we can assume that.

Either way, I should be wrapping the logic in a case statement so we can actually preempt the subquery unless it's singleton-queue, something like:

```sql
WHERE ... AND (
    CASE WHEN {{I am a singleton queue key}} THEN notExistsSubquery
    ELSE true
    END
)
```

Do you still want the new option or is the case statement sufficient?

Reply:
Can you verify that CASE will conditionally change the query plan?  I was suggesting `includeMetadata` since this a client-side technique that changes the query before sending to postgres.

Reply:
Ok, just pushed up those query changes. Here is the new fetch query's CTE when fetching a job channel that uses singleton queue (note I turned off sequential scans for these to force index usage):
```
Limit  (cost=230.53..230.53 rows=1 width=84) (actual time=0.081..0.083 rows=1 loops=1)
   InitPlan 1 (returns $0)
     ->  Bitmap Heap Scan on job active_job  (cost=132.89..222.36 rows=1 width=0) (actual time=0.062..0.062 rows=0 loops=1)
           Recheck Cond: (name IS NOT NULL)
           Filter: ((singletonkey IS NOT NULL) AND (state = 'active'::pgboss.job_state))
           Rows Removed by Filter: 83
           Heap Blocks: exact=40
           ->  Bitmap Index Scan on job_name  (cost=0.00..132.89 rows=83 width=0) (actual time=0.012..0.012 rows=83 loops=1)
                 Index Cond: (name IS NOT NULL)
   ->  Sort  (cost=8.17..8.17 rows=1 width=84) (actual time=0.081..0.081 rows=1 loops=1)
         Sort Key: job.priority DESC, job.createdon, job.id
         Sort Method: quicksort  Memory: 25kB
         ->  Index Scan using job_fetch on job  (cost=0.13..8.16 rows=1 width=84) (actual time=0.074..0.075 rows=1 loops=1)
               Index Cond: ((name = 'jobUsingSingletonQueue'::text) AND (startafter < now()))
               Filter: ((name ~~ 'jobUsingSingletonQueue'::text) AND CASE WHEN ((singletonkey IS NOT NULL) AND (singletonkey ~~ '\_\_pgboss\_\_singleton\_queue%'::text)) THEN (NOT $0) ELSE true END)
 Planning Time: 0.231 ms
 Execution Time: 0.111 ms
```
And here is the same CTE when fetching a job channel that does not use singleton queue (note the "never executed" annotations):
```
Limit  (cost=230.53..230.53 rows=1 width=84) (actual time=0.008..0.008 rows=0 loops=1)
   InitPlan 1 (returns $0)
     ->  Bitmap Heap Scan on job active_job  (cost=132.89..222.36 rows=1 width=0) (never executed)
           Recheck Cond: (name IS NOT NULL)
           Filter: ((singletonkey IS NOT NULL) AND (state = 'active'::pgboss.job_state))
           ->  Bitmap Index Scan on job_name  (cost=0.00..132.89 rows=83 width=0) (never executed)
                 Index Cond: (name IS NOT NULL)
   ->  Sort  (cost=8.17..8.17 rows=1 width=84) (actual time=0.007..0.008 rows=0 loops=1)
         Sort Key: job.priority DESC, job.createdon, job.id
         Sort Method: quicksort  Memory: 25kB
         ->  Index Scan using job_fetch on job  (cost=0.13..8.16 rows=1 width=84) (actual time=0.004..0.004 rows=0 loops=1)
               Index Cond: ((name = 'jobNotUsingSingletonQueue'::text) AND (startafter < now()))
               Filter: ((name ~~ 'jobNotUsingSingletonQueue'::text) AND CASE WHEN ((singletonkey IS NOT NULL) AND (singletonkey ~~ '\_\_pgboss\_\_singleton\_queue%'::text)) THEN (NOT $0) ELSE true END)
 Planning Time: 0.171 ms
 Execution Time: 0.030 ms
```
For reference, here's the CTE without any of my changes:
```
Limit  (cost=8.16..8.17 rows=1 width=84) (actual time=0.010..0.010 rows=0 loops=1)
   ->  Sort  (cost=8.16..8.17 rows=1 width=84) (actual time=0.009..0.010 rows=0 loops=1)
         Sort Key: priority DESC, createdon, id
         Sort Method: quicksort  Memory: 25kB
         ->  Index Scan using job_fetch on job  (cost=0.13..8.15 rows=1 width=84) (actual time=0.006..0.006 rows=0 loops=1)
               Index Cond: ((name = 'jobNotUsingSingletonQueue'::text) AND (startafter < now()))
               Filter: (name ~~ 'jobNotUsingSingletonQueue'::text)
 Planning Time: 0.151 ms
 Execution Time: 0.028 ms
```

Reply:
Thus, should be effectively no performance difference for non-singleton queue jobs. For singleton queue jobs, the subquery will run, again I think _not_ performing this check is a bug or at least not something people are relying on.

But I'm still happy to include the client-side option to essentially "feature flag" it if you want to limit risk, lemme know.

Reply:
I've looked at the plans on my side as well and see similar results.  This will have the side effect of doubling the current execution times on queries, which will result in a significant performance hit on a large tables.

Reply:
When I tested again w/ 3 million jobs in the table, I only saw an execution time slowdown of 0.03 ms per fetch (from ~0.06 to 0.09). Tho there is a planning time increase of ~0.1 ms.

But fair enough. I went ahead and created the `enforceSingletonQueueActiveLimit` option with usage similar to `includeMetadata`, as you suggested.

I included the performance caveat as part of the documentation updates.